docs/page-objects/inputs.rst

```diff
@@ -0,0 +1,65 @@
+.. _inputs:
+
+======
+Inputs
+======
+
+:ref:`Page object classes <page-object-classes>`, in their ``__init__`` method,
+must define input parameters with type hints pointing to input classes.
+
+Those input classes may be:
+
+-   :ref:`Built-in web-poet input classes <built-in-inputs>`.
+
+-   :ref:`Custom input classes <custom-inputs>`.
+
+-   Other :ref:`page object classes <page-object-classes>`.
+
+-   :ref:`Item classes <items>`, when using a :ref:`framework <frameworks>`
+    that can provide item classes.
+
+Based on the target URL and parameter type hints, :ref:`frameworks
+<frameworks>` automatically build the required objects at run time, and pass
+them to the ``__init__`` method of the corresponding page object class.
+
+For example, if a page object class has an ``__init__`` parameter of type
+:class:`~web_poet.page_inputs.HttpResponse`, and the target URL is
+https://example.com, your framework would send an HTTP request to
+https://example.com, download the response, build an
+:class:`~web_poet.page_inputs.HttpResponse` object with the response data, and
+pass it to the ``__init__`` method of the page object class being used.
+
+.. _built-in-inputs:
+
+Built-in input classes
+======================
+
+The :mod:`web_poet.page_inputs` module defines multiple classes that you can
+define as inputs for a page object class, including:
+
+-   :class:`~web_poet.page_inputs.HttpResponse`, a complete HTTP response,
+    including URL, headers, and body. This is the most common input for a page
+    object class.
```

Comment:
Would it make sense to mention BrowserHtml as well? It looks different from other classes (e.g. HttpResponseBody), because it's not covered by HttpResponse.

Reply:
So, is the plan to add it here once we implement BrowserHtml support e.g. in scrapy-zyte-api?